HelpSource/Classes/NotificationCenter.schelp

```diff
@@ -4,9 +4,35 @@ related:: Classes/SimpleController, Classes/NodeWatcher
 categories:: Control
 
 description::
-One common OOP pattern is Model-View-Controller where one object (the controller) is a dependant of the model. Every time the model changes it notifies all of its dependants. In this case the model has a dictionary of dependants and iterates through those.
+One common OOP pattern is the Observer Pattern where one object (the controller or observer) is a dependant of a model or subject object ( https://en.wikipedia.org/wiki/Observer_pattern ). Every time the model changes it notifies all of its dependants. In the default implementation of this pattern in SuperCollider, the model has a dictionary of dependants and iterates through those.  A limitation of this implementation is that all dependants always receive the same message "update" when the model changes.  If a dependant is required to perform different actions depending on the type of change that happens in the model, or on the dependant itself, then this must be coded in the update method of the dependant.  The NotificationCenter overcomes this limitation by making it possible to register a function to be performed for each combination of model, notification message and dependant.  There are thus 4 objects involved in each connecetion between model and dependant in the NotificationCenter:
```

Comment:
i think this could be easier to read as multiple paragraphs.

Reply:
also, there seems to be considerable confusion in this help file due to use of multiple names for the same thing. it's model/dependant in the help file, but object/listener in the actual argument names. i'd suggest always writing "object" and "listener" to keep it consistent.

Reply:
GOF book uses `Observer` and `Subject`, would be good to use their terms?

Reply:
imo it would be confusing to use one convention in the help file and a different one in the actual API, which is "object" and "listener". it's unfortunate that the API seems to deviate from standard terminology (and can't be changed due to backward compatibility) but it's most important to be internally consistent.

Reply:
Thanks for the comments. I will split explanation to multiple paragraphs and rename the argument names for consistency as suggested. I agree with @snappizz that internal consistency has priority. References to GOF terminology can be provided in the present help documentation, and I will try to do this.